examples/quantum1d/7.xy-finiteT/main.jl

```diff
@@ -0,0 +1,471 @@
+Keeping these parameters in balance is necessary to obtain accurate results, and this might require some trial and error.
+"""
+
+md"""
+### MPO multiplication approach (exponential)
+
+If we wish to push the results to even larger $\beta$ values, we can note that taking linear steps in $\beta$ is not the only option.
+To that end, we can use another trick to scan over an exponential range of $\beta$ values: [exponentiating by squaring](https://en.wikipedia.org/wiki/Exponentiation_by_squaring).
+In particular, we note that computing $x^n$ for integer (large) $n$ can typically be done more efficiently than computing $x * x * ... * x$.
+To do so, we note that multiplication is associative, and regroup the factors in such a way that we can compute the result in a logarithmic number of steps.
+Here, we assume $n = 2^m$ for some integer $m$, and note that this could be generalized to any $n$ by decomposing $n$ into a sum of powers of $2$.
+Then, we can write
+
+```math
+x^n = x^{2^m} = x^{2^{m-1}} \cdot x^{2^{m-1}} = (x^{2^{m-2}} \cdot x^{2^{m-2}}) \cdot (x^{2^{m-2}} \cdot x^{2^{m-2}}) = \dots
+```
+
+In other words, we can scan an exponential range of $\beta$ values by squaring the density matrix at each step.
+"""
+
+βs_exp = 2.0 .^ (-3:3)
+Z_mpo_mul_exp = zeros(length(βs_exp))
+
+## first iteration: start from high order Taylor expansion
+ρ₀ = make_time_mpo(H, im * first(βs_exp) / 2, TaylorCluster(; N=3))
+Z_mpo_mul_exp[1] = real(dot(ρ₀, ρ₀))^(1 / N)
+
+## subsequent iterations: square
+ρ = ρ₀
+ρ_mps = convert(FiniteMPS, ρ₀)
+for i in 2:length(βs_exp)
+    global ρ_mps, ρ
+    @info "Computing β = $(βs_exp[i])"
+    ρ_mps, = approximate(ρ_mps, (ρ, ρ_mps),
+                         DMRG2(; trscheme=truncdim(D_max), maxiter=10))
+    Z_mpo_mul_exp[i] = real(dot(ρ_mps, ρ_mps))^(1 / N)
+    ρ = convert(FiniteMPO, ρ_mps)
+end
+F_mpo_mul_exp = -(1 ./ βs_exp) .* log.(Z_mpo_mul_exp)
+
+p_mpo_mul_exp_diff = let
+    labels = reshape(map(expansion_orders[2:end]) do N
+                         return "Taylor N=$N"
+                     end, 1, :)
+    p1 = plot(βs, abs.(real.(Z_taylor2) .- partition_function.(βs, J, N));
+              label=labels, title="Partition function error",
+              xlabel="β", ylabel="ΔZ(β)")
+    plot!(p1, βs, abs.(real.(Z_mpo_mul) .- partition_function.(βs, J, N));
+          label="MPO multiplication")
+    plot!(p1, βs_exp, abs.(real.(Z_mpo_mul_exp) .- partition_function.(βs_exp, J, N));
+          label="MPO multiplication exp")
+
+    p2 = plot(βs, abs.(real.(F_taylor2) .- free_energy.(βs, J, N)); label=labels,
+              xlabel="β", ylabel="ΔF(β)", title="Free energy error")
+    plot!(p2, βs, abs.(real.(F_mpo_mul) .- free_energy.(βs, J, N));
+          label="MPO multiplication")
+    plot!(p2, βs_exp, abs.(real.(F_mpo_mul_exp) .- free_energy.(βs_exp, J, N));
+          label="MPO multiplication exp")
+    plot(p1, p2)
```

Comment:
Same comment about the legend in the right panel, would be better in the northwest corner.